ChemistryLib/PhreeqcIO.cpp

```diff
@@ -139,12 +142,18 @@ void PhreeqcIO::doWaterChemistryCalculation(
 void PhreeqcIO::setAqueousSolutionsOrUpdateProcessSolutions(
     std::vector<GlobalVector*> const& process_solutions, Status const status)
 {
-    std::size_t const num_chemical_systems = _aqueous_solutions.size();
+    std::size_t const num_chemical_systems = _mesh.getNumberOfBaseNodes();
+
+    auto const chemical_system_map =
+        *_mesh.getProperties().template getPropertyVector<std::size_t>(
+            "bulk_node_ids", MeshLib::MeshItemType::Node, 1);
```

Comment:
The default bulk mesh will not have the `bulk_node_ids`. (And it should not.)

Reply:
I haven't realized.

Reply:
=> The chemical domain mesh must always be a subdomain mesh (not necessarily smaller).

Maybe it would be better to have a mesh property indicating where the chemical system should be solved?

Reply:
Good point! One need apply the mesh tool `identifySubdomain` to create a mesh file containing the property `bulk_node_ids`, even in the case that chemical reactions take place over the entire domain. Also, the tag `<mesh>` should no longer be optional.